File: regex.scala
#!/usr/bin/env scala -classpath google-gson-2.2.4/gson-2.2.4.jar

import com.google.gson.Gson

val gson = new Gson()
val string = args(0)
var flags = ""
try { flags = args(2) } catch { case e: Exception => }

var regex = if(flags contains "i")
              "(?i)" + args(1)
            else
              args(1)

val json = gson.toJson(
             if(flags contains "g")
               regex.r.findAllIn(string).toArray
             else
               regex.r.findFirstIn(string).toArray
           )

val result = if(json == "[]") "null" else json

println(result)

File: regex.rb
#!/usr/bin/env ruby

require 'json'

string = ARGV[0] || ''
regex = ARGV[1] || ''
flags = ARGV[2] || ''

# puts 'string: ' + string
# puts 'regex: ' + regex
# puts 'flags: ' + flags

case_insensitive = flags.include?('i') ? true : false
result = if flags.include?('g')
    string.scan(Regexp.new(regex, case_insensitive)).to_json
  else
    [string.match(Regexp.new(regex, case_insensitive))].to_json
  end

# massage results to be like nodejs
result = 'null' if ['[]', '[null]'].include? result

puts result
